modules/base/rendering/directionallightsource.cpp

```diff
@@ -0,0 +1,236 @@
+ *                                                                                       *
+ * Copyright (c) 2014-2025                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/directionallightsource.h>
+
+#include <modules/base/basemodule.h>
+#include <modules/base/rendering/renderablemodel.h>
+#include <openspace/engine/globals.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/scene/scene.h>
+#include <openspace/scene/scenegraphnode.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/opengl/programobjectmanager.h>
+#include <limits>
+
+namespace {
+
+    constexpr std::string_view _loggerCat = "DirectionalLightsource";
+
+    constexpr int DepthMapResolutionMultiplier = 4;
+    constexpr double ShadowFrustumDistanceMultiplier = 500.0;
+
+    struct [[codegen::Dictionary(DirectionalLightsource)]] Parameters {
+
+    };
+#include "directionallightsource_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation DirectionalLightSource::Documentation() {
+    return codegen::doc<Parameters>("base_renderable_directionallightsource");
+}
+
+DirectionalLightSource::DirectionalLightSource(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+{
+    setRenderBin(Renderable::RenderBin::Background);
```

Comment:
So we render the shadows first? I would intuitively think we would render them last?